modules/store/src/reducer_creator.ts

```diff
@@ -39,15 +53,29 @@ export interface OnReducer<State, Creators extends readonly ActionCreator[]> {
  * on(AuthApiActions.loginSuccess, (state, { user }) => ({ ...state, user }))
  * ```
  */
-export function on<State, Creators extends readonly ActionCreator[]>(
+export function on<
+  // State type that is being passed from `createReducer` when created within that factory function
+  State,
+  // Action creators
+  Creators extends readonly ActionCreator[],
+  // Inferred type from within OnReducer function if `State` is unknown. This is typically the case when `on` function
+  // is created outside of `createReducer` and state type is either explicitly set OR inferred by return type.
+  // e.g. `const onFn = on(action, (state: State, {prop}) => ({ ...state, name: prop }));`
```

Comment:
Thanks for adding the great comments! Minor nit, using `e.g.`

```suggestion
  // For example: `const onFn = on(action, (state: State, {prop}) => ({ ...state, name: prop }));`
```